rosidl_parser/rosidl_parser/definition.py

```diff
@@ -56,49 +101,64 @@
 
 
 class AbstractType:
-    """The base class for all types."""
+    """The abstract base class for all types."""
 
     __slots__ = ()
 
     def __eq__(self, other):
         return type(self) == type(other)
 
 
-class BaseType(AbstractType):
-    """The base class for types which can be used inside nested types."""
+class AbstractNestableType(AbstractType):
+    """
+    The abstract base class for types which can be used inside nested types.
+
+    Arrays and sequences are nested types which contain elements of nestable
+    types.
+    Nestable types are:
+    - BasicType like numerics, character types, boolean and octet
+    - NamedType identified by an name which hasn't been resolved yet
+    - NamespacedType which describe another Structure
+    - Strings with any kind of character types, bounded as well as unbounded
+    """
 
     __slots__ = ()
 
 
-class BasicType(BaseType):
-    """A basic type according to the IDL specification."""
+class BasicType(AbstractNestableType):
+    """
+    A basic type according to the IDL specification.
+
+    Its `typename` attribute defines the specific numeric, character, boolean
+    and octet type.
+    """
 
-    __slots__ = ('type', )
+    __slots__ = ('typename', )
```

Comment:
Instead of using the generic name `type` for multiple slots I changed the slot names to something more specific. Instead of accessing something like `foo.type.type.type` it will be more readable like `foo.structure.namespaced_type.typename` (just a made up example).